src/YUILoader.cc

```diff
@@ -58,74 +76,88 @@ void YUILoader::loadUI( bool withThreads )
     isGtk = ( ( strstr( envDesktop, "Unity"    ) != NULL ) || isGtk );
     isGtk = ( ( strstr( envDesktop, "XFCE"     ) != NULL ) || isGtk );
 
-    if( isGtk )
-    {
-	yuiMilestone () << "Detected a Gtk-based desktop environment." << std::endl;
-	yuiMilestone () << "Prefering Gtk-UI if available." << std::endl;
-    }
+    if( isGtk ) yuiMilestone () << "Detected a Gtk-based desktop environment." << std::endl
+                                << "Prefering Gtk-UI if available and no" << std::endl
+                                << "user-selected override is present." << std::endl;
 
     YCommandLine cmdline;
 
-    bool wantNcurses = cmdline.find("--ncurses") != -1;
-    if( wantNcurses )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_NCurses << "\". Forced on command-line." << std::endl;
-    bool wantQt = cmdline.find("--qt") != -1;
-    if( wantQt )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_Qt << "\". Forced on command-line." << std::endl;
-    bool wantGtk = cmdline.find("--gtk") != -1;
-    if( wantGtk )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_Gtk << "\". Forced on command-line." << std::endl;
+    bool wantGtk     = ( cmdline.find( "--gtk" )     != -1 );
+    bool wantNcurses = ( cmdline.find( "--ncurses" ) != -1 );
+    bool wantQt      = ( cmdline.find( "--qt" )      != -1 );
+    wantGtk     = ( wantGtk     || ( strcmp( envPreset, YUIPlugin_Gtk )     == 0 ) ) && !( wantNcurses || wantQt );
+    wantNcurses = ( wantNcurses || ( strcmp( envPreset, YUIPlugin_NCurses ) == 0 ) ) && !( wantGtk     || wantQt );
+    wantQt      = ( wantQt      || ( strcmp( envPreset, YUIPlugin_Qt )      == 0 ) ) && !( wantGtk     || wantNcurses );
```

Comment:
Yes, didn't see the easier way, when I coded this…  I fixed it in the rebased commits.